Plugins/Player/Player.cpp

```diff
@@ -1428,4 +1430,30 @@ ArgumentStack Player::SetCreatureNameOverride(ArgumentStack&& args)
     return Services::Events::Arguments();
 }
 
+ArgumentStack Player::FloatingTextStringOnCreature(ArgumentStack&& args)
+{
+    if (auto *pPlayer = player(args))
+    {
+        auto oidCreature = Services::Events::ExtractArgument<ObjectID>(args);
+          ASSERT_OR_THROW(oidCreature != Constants::OBJECT_INVALID);
+        auto text = Services::Events::ExtractArgument<std::string>(args);
+          ASSERT_OR_THROW(!text.empty());
+
+        if (auto *pCreature = Utils::AsNWSCreature(Utils::GetGameObject(oidCreature)))
+        {
+            if (auto *pMessage = static_cast<CNWSMessage*>(Globals::AppManager()->m_pServerExoApp->GetNWSMessage()))
+            {
+                auto *pMessageData = new CNWCCMessageData();
+                pMessageData->SetObjectID(0, pCreature->m_idSelf);
+                pMessageData->SetInteger(9, 94);
+                pMessageData->SetString(0, text);
+
+                pMessage->SendServerToPlayerCCMessage(pPlayer->m_nPlayerID, 11, pMessageData, nullptr);
```

Comment:
Not sure if `pMessageData` needs a `delete` after sending it.

Reply:
Digging around some more, it appears it does need one.

Reply:
Read in clippy's voice: If it's deleted when it goes out of this scope, you can stack allocate with `CNWCCMessageData messageData;` instead of using a raw pointer.

Reply:
Oh, that's even better, thanks!